### ISSUE TITLE
Extending an arbitary expression that can be null at run time

### ISSUE THREAD

**Issue by yuit** [Microsoft/TypeScript#4049]
In this original issue #3696. Currently, we give an error in the following case
```typescript
class A extends null {
   constructor() {
      super();
   }
```

However, in the case of an arbitrary expression that can return null, we will not give an appropriate error and will also give an error for not calling ```super```
There are possible fixes for the problem:
* Do not give any error for not calling ```super``` when users extends an arbitrary expression.
* Only allow extend ```null``` explicitly
* Provide a compiler directive for users to suppress error for not calling ```super```

**Comment by mhegazy** [comment]
I would say do nothing.

**Comment by DanielRosenwasser** [comment]
Either that or @yuit's first suggestion.

**Comment by mhegazy** [comment]
this should be fixed now.
